From d720d5ecb0d9944c77683b9bbffc9ff44ae8eb87 Mon Sep 17 00:00:00 2001
From: xiaoming wang <xiaoming.wang@intel.com>
Date: Sun, 19 Aug 2012 17:20:37 -0400
Subject: [PORT FROM MAIN] User Stack: User space stack is not integrated

BZ: 51255

The upgrade of gcc compiler cause register ebp used as normal register
because compile option -fomit-frame-pointer used as default option
in compiler. So should add compile option  -fno-omit-frame-pointer
to dump stack integrated in user space.

Change-Id: Icc1d6eef9541f2ed94ca1b335244e1de5ffb2231
Orig-Change-Id: Iaa254383ada88e7728d4b3ebe720da6684777d26
Signed-off-by: xiaoming wang <xiaoming.wang@intel.com>
Signed-off-by: xindong ma<xindong.ma@intel.com>
Signed-off-by: Christophe Guerard <christophe.guerard@intel.com>
---
 core/combo/TARGET_linux-x86.mk |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index 5bb5ad6..9632fe2 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -86,6 +86,10 @@ else
 endif
 KERNEL_HEADERS := $(KERNEL_HEADERS_COMMON) $(KERNEL_HEADERS_ARCH)
 
+ifneq ($(TARGET_BUILD_VARIANT),user)
+TARGET_GLOBAL_CFLAGS += -fno-omit-frame-pointer
+endif
+
 TARGET_GLOBAL_CFLAGS += \
 			-O2 \
 			-Ulinux \
-- 
1.7.4.1

